src/backend/distributed/worker/worker_create_or_replace.c

```diff
@@ -68,6 +69,8 @@ WrapCreateOrReplace(const char *sql)
  *  - non destructive if there is data store that would be destroyed if the object was
  *    used in a table on this node, eg. types. If the type would be dropped with a cascade
  *    it would drop any column holding user data for this type.
+ *
+ *  TODO: we can probably remove the existing object if it has no dependents
```

Comment:
This deserves a bit of a deeper discussion.

I feel like we need to refactor `worker_create_or_replace_object` in a way that it can optionally take a list of statements and reconciles the object if possible.

`TEXT SEARCH CONFIGRURATION` can be created as follows:

```sql
CREATE TEXT SEARCH CONFIGURATION foo ( copy = dutch );
```

From the catalog the relation to the `pg_catalog.dutch` text search configuration is lost. Instead it used said configuration to copy its parser and all its mappings. Hence we will create the object on the workers like the following:

```
CREATE TEXT SEARCH CONFIGURATION public.foo ( parser = pg_catalog.default );
ALTER TEXT SEARCH CONFIGURATION public.foo ADD MAPPING FOR token1 WITH dict1, dict2;
ALTER TEXT SEARCH CONFIGURATION public.foo ADD MAPPING FOR token2 WITH dict3, dict;
...
```

All statements combined form the state of the object.
Ideally we would verify all statements are the same between coordinator and worker and do nothing (except mark them distributed eventually). This _should_ work well for all configurations installed with initdb as well.

If settings differ we could still fallback on old behaviour to rename the old object and recreate according to spec.
And maybe drop the object if it wasn't used anyway, like the TODO describes.

The dropping is trivial, but not related to this PR, so might perform this in a separate PR.
The refactor of worker replace is actually related to this PR, but sizeable work, although I think a requirement to _not_ pollute catalogs on the workers when initdb created configs are used in tables/indices.

Reply:
a potential way to prevent initdb created configuration to be overwritten with semantically the same object: https://github.com/citusdata/citus/pull/5703